### ISSUE TITLE
Admin user should be able to clone/push to all repositories

### ISSUE THREAD

**Issue by xor-gate** [gogits/gogs#2452]
I think the admin user should be able to clone and push to all repositories without being in a organisation and team.

This is somewhat related to this issue: https://github.com/gogits/gogs/issues/1146

**Comment by bkcsoft** [comment]
give me a valid use-case and I'll give you a PR :laughing:

on a more serious note, why would you want this? I get that admins should be able to remove/disable offending repos/users, but commit-access? :weary:

**Comment by xor-gate** [comment]
@bkcsoft, you are probably right about the commit access. Better would be to have only read-only (clone) access then. Because the admin is able to fork even private repos where he is not in the organisation, so read access is indirectly allowed.

**Comment by nanoant** [comment]
I do agree that admins should be able to clone the repo for some Gogs installations, eg. to investigate if repo contains some malware or infringes some copyrights. If Gogs admin is server admin too, this will make the task just easier.

But agree with @bkcsoft that admins should NOT have commit access. They don't own the code, so should not be given implicit permissions to alter it, at least not without other repo users/owner consent.
If admin wants to alter the code, may add him/herself to the org members or collaborators, which lets the others know what admin is gonna to do.

**Comment by xor-gate** [comment]
This also means that admins should be able to fork all repos and create a pull request for changes.

**Comment by nanoant** [comment]
@xor-gate Yup, being able to clone all repos and make PRs if you're admin makes sense.
